File: Iteration 1/code/CSE3063GRP12/src/features/weekly_schedule/WeeklyScheduleController.java
package features.weekly_schedule;

import java.io.IOException;
import java.util.ArrayList;

import core.models.concretes.Course;
import core.repositories.CourseRepository;

import core.general_providers.SessionController;
import core.general_providers.TerminalManager;
import features.main_menu.MenuController;
import core.models.concretes.Student;

public class WeeklyScheduleController {

    private WeeklyScheduleView weeklyScheduleView;
    private CourseRepository courseRepository;

    public WeeklyScheduleController() {
        this.weeklyScheduleView = new WeeklyScheduleView();
        this.courseRepository = new CourseRepository();
        handleWeeklySchedule();
    }

    private ArrayList<Course> fetchCourses() throws IOException {

        // UNCOMMENT:
        Student currentStudent = (Student) (SessionController.getInstance().getCurrentUser());
        int currentSemester = currentStudent.getTranscript().getCurrentSemester();
        return courseRepository.getCoursesBySemester(currentSemester);

    }

    private void navigateToMenu() {
        new MenuController();
    }

    private String getUserInput() {
        weeklyScheduleView.showQuitMessage();
        return TerminalManager.getInstance().read();
    }

    private void handleWeeklySchedule() {
        try {
            ArrayList<Course> currentCourses = fetchCourses();
            weeklyScheduleView.showWeeklySchedule(currentCourses);
            if (getUserInput().equals("q"))
                navigateToMenu();
        } catch (Exception e) {
            e.printStackTrace();
        }

    }
}

File: Iteration3/code/CSE3063GRP12/src/features/student/weekly_schedule/WeeklyScheduleController.py
from features.student.weekly_schedule.WeeklyScheduleView import WeeklyScheduleView
from core.repositories.CourseRepository import CourseRepository
from core.general_providers.SessionController import SessionController
from core.exceptions.UnexpectedInputException import UnexpectedInputException

class WeeklyScheduleController:
    def __init__(self):
        self.weekly_schedule_view = WeeklyScheduleView()
        self.course_repository = CourseRepository()
        self.handle_weekly_schedule()

    def fetch_courses(self):
        current_student = SessionController.getInstance().get_current_user()
        current_semester = current_student.transcript.get("currentSemester")
        return self.course_repository.get_courses_by_semester(current_semester)

    def navigate_to_menu(self):
        from features.main_menu.MenuController import MenuController
        MenuController()

    def handle_weekly_schedule(self):
        try:
            current_courses = self.fetch_courses()
            self.weekly_schedule_view.show_weekly_schedule(current_courses)
            input_message = input("Press q to return Main Menu: ")
            if input_message == "q":
                self.navigate_to_menu()
            else:
                raise UnexpectedInputException("Unexpected input!")
        except Exception as e:
            print(e)